### PR TITLE
Add signing with Developer ID

### LINKED ISSUE #13: Sign Compiler with Apple
Screen Shot 2019-10-17 at 5 38 46 PM

We should sign the tools with apple to make the user experience more seamless: https://developer.apple.com/news/?id=09032019a

### PR DESCRIPTION
**Pull request by prateekma**
Closes #13
The resulting .tar.gz was zipped and sent to Apple for notarization. The notarization was successful. I also copied everything to `~/.gradle/toolchains/frc/2020/` and compiled a test project:

```
$ ./gradlew clean build
Skipping builds for bionic (toolchain is marked optional)
Skipping builds for raspbian (toolchain is marked optional)

> Task :compileJava
warning: unknown enum constant Visibility.NONE
  reason: class file for com.fasterxml.jackson.annotation.JsonAutoDetect$Visibility not found
warning: unknown enum constant Visibility.NONE
warning: unknown enum constant Visibility.NONE
3 warnings

> Task :javadoc
warning: unknown enum constant Visibility.NONE
  reason: class file for com.fasterxml.jackson.annotation.JsonAutoDetect$Visibility not found
warning: unknown enum constant Visibility.NONE
warning: unknown enum constant Visibility.NONE
/Users/prateekma/Documents/Dev/photonlib/src/main/java/org/photonvision/Packet.java:34: warning: no @param for size
  public Packet(int size) {
         ^
4 warnings

> Task :installCppTestLinuxathenaDebugGoogleTestExe
/Users/prateekma/Documents/Dev/photonlib/build/exe/cppTest/linuxathena/debug/cppTest
cppT

> Task :compileTestJava
warning: unknown enum constant Visibility.NONE
  reason: class file for com.fasterxml.jackson.annotation.JsonAutoDetect$Visibility not found
warning: unknown enum constant Visibility.NONE
warning: unknown enum constant Visibility.NONE
3 warnings

> Task :generateVendorJson
Writing version 2020.0.7 to /Users/prateekma/Documents/Dev/photonlib/build/generated/vendordeps/photonlib.json

> Task :installCppTestLinuxathenaReleaseGoogleTestExe
/Users/prateekma/Documents/Dev/photonlib/build/exe/cppTest/linuxathena/release/cppTest
cppT

> Task :installCppTestOsxx86-64ReleaseGoogleTestExe
/Users/prateekma/Documents/Dev/photonlib/build/exe/cppTest/osxx86-64/release/cppTest
cppT

> Task :runCppTestOsxx86-64ReleaseGoogleTestExe
[==========] Running 3 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 3 tests from PacketTest
[ RUN      ] PacketTest.PhotonTrackedTarget
64,8,0,0,0,0,0,0,64,16,0,0,0,0,0,0,64,34,0,0,0,0,0,0,-64,20,0,0,0,0,0,0,63,-16,0,0,0,0,0,0,64,0,0,0,0,0,0,0,64,85,124,101,19,-54,-47,122,[       OK ] PacketTest.PhotonTrackedTarget (1 ms)
[ RUN      ] PacketTest.PhotonPipelineResult
[       OK ] PacketTest.PhotonPipelineResult (0 ms)
[ RUN      ] PacketTest.BytePackFromJava
[       OK ] PacketTest.BytePackFromJava (0 ms)
[----------] 3 tests from PacketTest (1 ms total)

[----------] Global test environment tear-down
[==========] 3 tests from 1 test suite ran. (1 ms total)
[  PASSED  ] 3 tests.

> Task :installCppTestOsxx86-64DebugGoogleTestExe
/Users/prateekma/Documents/Dev/photonlib/build/exe/cppTest/osxx86-64/debug/cppTest
cppT

> Task :runCppTestOsxx86-64DebugGoogleTestExe
[==========] Running 3 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 3 tests from PacketTest
[ RUN      ] PacketTest.PhotonTrackedTarget
64,8,0,0,0,0,0,0,64,16,0,0,0,0,0,0,64,34,0,0,0,0,0,0,-64,20,0,0,0,0,0,0,63,-16,0,0,0,0,0,0,64,0,0,0,0,0,0,0,64,85,124,101,19,-54,-47,122,[       OK ] PacketTest.PhotonTrackedTarget (0 ms)
[ RUN      ] PacketTest.PhotonPipelineResult
[       OK ] PacketTest.PhotonPipelineResult (0 ms)
[ RUN      ] PacketTest.BytePackFromJava
[       OK ] PacketTest.BytePackFromJava (0 ms)
[----------] 3 tests from PacketTest (0 ms total)

[----------] Global test environment tear-down
[==========] 3 tests from 1 test suite ran. (1 ms total)
[  PASSED  ] 3 tests.

BUILD SUCCESSFUL in 33s
59 actionable tasks: 58 executed, 1 up-to-date
```

### CODE REVIEW

**Comment by prateekma** [comment]
https://github.com/prateekma/roborio-toolchain/runs/1272328789?check_suite_focus=true The resulting macOS artifact has the signed binaries.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,6 +39,27 @@ jobs:
           name: macOS
           path: mac/*Toolchain*.tar.gz
 
+  sign-macos:
```
Comment (AustinShalit) [inline .github/workflows/ci.yml]: I think I would rather have this be a part of the build-macos job and the signing steps having an `if` to only run on master/tags

**Review by RyanHir — COMMENTED** [review]
Correct me if I am wrong but I believe `frc2020/roborio/arm-frc2020-linux-gnueabi/bin` is the sysroot of the RoboRio and is never executed but is used for linking.

Same for (but less confident) with `frc2020/roborio/libexec/gcc/arm-frc2020-linux-gnueabi/7.3.0`

All the binaries that need signing are in `frc2020/roborio/bin`

File: codesign.sh [hunk 2]
```diff
@@ -0,0 +1,17 @@
+sign_directory()
```
Comment (RyanHir) [inline codesign.sh:3]: Perhaps add `#! /bin/bash` to the top to tell the environment to use the correct shell as it is possible (but unlikly) that another default shell can change how it is interpreted.

**Comment by prateekma** [comment]
@RyanHir We submitted the entire archive for notarization and it complained about unsigned binaries in those three directories.

**Comment by RyanHir** [comment]
ah, thanks. I did not realize it wanted the arm binaries

**Comment by prateekma** [comment]
The notarization service says they are x86_64 binaries: https://hastebin.com/avozonujig.kotlin

**Comment by RyanHir** [comment]
Interesting, I would have thought that the first directory was a sysroot

File: codesign.sh [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#!/bin/sh
```
Comment (RyanHir) [inline codesign.sh]: I would be careful with this, it should be fine on MacOS. But it is not in the POSIX standard to have `[ ]` testing but is in natively for bash.

**Review by RyanHir — DISMISSED** [review]

**Review by PeterJohnson — APPROVED** [review]

**Pull request merged**